lib/Unary.fz

```diff
@@ -0,0 +1,39 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature Unary
+#
+# -----------------------------------------------------------------------
+
+# Unary -- function that takes exactly one argument and returns a result
+#
+# T is the result type of the function, U is the argument type of the function
+#
+# For unary functions, function composition is possible if the result type of
+# the first function matches the argument type of the second function.
+#
+Unary(T, U type) ref : Function T U is
+
+
+  # function composition
+  #
+  infix ∘ (V type, other Unary U V) Unary T V is
+    ref : Unary T V
+      redef call (v V) T is
+        Unary.this.call (other.call v)
```

Comment:
```
v -> call (other.call v) 
```
should work as an implementation as well, maybe even 
```
v -> Unary.this (other v) 
```

Reply:
The type of the outer `call` is `T`, but the type of `v -> call..` should then be `Unary T V`. Anyway, this should be reproducible in a small test, can you check and create issues for these two cases?